examples/llava/clip.h

```diff
@@ -71,10 +76,12 @@ CLIP_API bool clip_image_load_from_bytes(const unsigned char * bytes, size_t byt
 /** preprocess img and store the result in res_imgs, pad_to_square may be overridden to false depending on model configuration */
 CLIP_API bool clip_image_preprocess(struct clip_ctx * ctx, const struct clip_image_u8 * img, struct clip_image_f32_batch * res_imgs );
 
+CLIP_API void uhd_normalize_image_u8_to_f32(struct clip_ctx * ctx, const clip_image_u8* src, clip_image_f32* dst);
+
 CLIP_API struct ggml_tensor * clip_get_newline_tensor(const struct clip_ctx * ctx);
 
-CLIP_API bool clip_image_encode      (struct clip_ctx * ctx, int n_threads, struct clip_image_f32 * img, float * vec);
-CLIP_API bool clip_image_batch_encode(struct clip_ctx * ctx, int n_threads, const struct clip_image_f32_batch * imgs, float * vec);
+CLIP_API bool clip_image_encode      (struct clip_ctx * ctx, int n_threads, struct clip_image_f32 * img, float * vec, struct load_image_size * load_image_size = nullptr);
+CLIP_API bool clip_image_batch_encode(struct clip_ctx * ctx, int n_threads, const struct clip_image_f32_batch * imgs, float * vec, struct load_image_size * load_image_size = nullptr);
```

Comment:
What is purpose of the `load_image_size` argument here? We already pass `img` or `imgs` that have sizes. How is `load_image_size` different from those sizes?

Reply:
Our method will perform different image slicing processes based on the image size, but when vit calculates pos_embed, it depends on the size of the original image. At this time, the image passed in has lost its original information, so a structure is needed to record the original image size and transmit it to the calculation layer.

I don't know if this is enough to answer your question. Feel free to ask me.

Reply:
Maybe the `clip_ctx` can keep this information to avoid passing it explicitly later on?

Reply:
Makes sense, let me think about it.

Reply:
[fcde997](https://github.com/ggerganov/llama.cpp/pull/7599/commits/fcde997126177763e1d091148108c941b73bf583)
I moved the "load_image_size" to "clip_ctx".